### DISCUSSION TITLE
Get past data with Replay

### DISCUSSION THREAD

**Discussion by cslimani** [Mathieu2301/TradingView-API#60] [General]
Is it possible get indicatosr data in the past as if I was using Replay ?

**Comment by Mathieu2301** [comment]
You can use the fake replay mode like [this](https://github.com/Mathieu2301/TradingView-API/blob/main/examples/FromToData.js), or like that:

```javascript
const TradingView = require('@mathieuc/tradingview');

const client = new Client();
const chart = new client.Session.Chart();

chart.setMarket('BINANCE:BTCEUR', {
  timeframe: '240',
  range: -10,
  to: 1600000000,
});

chart.onUpdate(() => {
  const periods = chart.periods
    .filter((p) => p.time > 1600012800)
    .map((p) => p.time);
  console.log(periods[0], periods[0] - periods[1]);
});

setInterval(() => {
  if (!chart.infos.full_name) return;
  chart.fetchMore(-1);
}, 1000);
```

I'm working on the real replay mode.

**Comment by Mathieu2301** [comment]
I pushed a beta implementation of the real replay mode. You can see an example [here](https://github.com/Mathieu2301/TradingView-API/blob/main/examples/ReplayMode.js).

**Reply by cslimani** [reply]
Thank you very much, that's exactly what I need.
